internal/net/tcp/control_unix.go

```diff
@@ -26,30 +26,35 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// TCP_FASTOPEN represents TCP_FASTOPEN.
 const TCP_FASTOPEN int = 0x17
 
+// Control controls raw network connection.
+// And Control calls c.Control invokes f on the underlying connection's file
+// descriptor or handle.
 func Control(network, address string, c syscall.RawConn) (err error) {
-	return c.Control(func(fd uintptr) {
-		f := int(fd)
-		ierr := syscall.SetsockoptInt(f, syscall.SOL_SOCKET, unix.SO_REUSEPORT, 1)
-		if ierr != nil {
-			err = errors.Wrap(err, ierr.Error())
-		}
-		ierr = syscall.SetsockoptInt(f, syscall.SOL_SOCKET, syscall.SO_REUSEADDR, 1)
-		if ierr != nil {
-			err = errors.Wrap(err, ierr.Error())
-		}
-		ierr = syscall.SetsockoptInt(f, syscall.IPPROTO_TCP, TCP_FASTOPEN, 1)
-		if ierr != nil {
-			err = errors.Wrap(err, ierr.Error())
-		}
-		ierr = syscall.SetsockoptInt(f, syscall.IPPROTO_TCP, syscall.TCP_NODELAY, 1)
-		if ierr != nil {
-			err = errors.Wrap(err, ierr.Error())
-		}
-		ierr = syscall.SetsockoptInt(f, syscall.IPPROTO_TCP, syscall.TCP_DEFER_ACCEPT, 1)
-		if ierr != nil {
-			err = errors.Wrap(err, ierr.Error())
+	errfn := func(errs ...error) (err error) {
```

Comment:
the problem of this refactoring is that it will not show the correct test coverage of this test case. 🤔 I am not sure should we do it.

Reply:
> I am not sure should we do it.
sorry, what is `it` ?

Reply:
@vankichi `it` means this refactoring